usage/cli/src/main/java/brooklyn/cli/Main.java

```diff
@@ -964,4 +979,35 @@ public ToStringHelper string() {
     protected Class<? extends BrooklynCommand> cliDefaultInfoCommand() {
         return DefaultInfoCommand.class;
     }
+    
+    public static class AppShutdownRequestListener implements ShutdownRequestListener {
+        private AtomicBoolean mutex = new AtomicBoolean();
+
+        @Override
+        public void onShutdownRequest() {
+            synchronized (mutex) {
```

Comment:
I prefer use of CountDownLatch - slightly simpler code, which is always good when dealing with multi-threading.